src/components/inputs/InputField.tsx

```diff
@@ -29,6 +34,14 @@ export default function InputField({
   minPasswordLength,
   dataTestId,
 }: InputProps) {
+  const [inputType, setInputType] = useState<string>(type);
```

Comment:
Since we make some assumptions & checks based on the type prop, I'd say we can type it in a more string way than just "string"

```suggestion
  const [inputType, setInputType] = useState<InputTypeEnum>(type);
```

where

```
export enum InputTypeEnum {
  PASSWORD = "password",
  TEXT = "text"
}
```

and the `InputProps` is converted to interface and type uses `InputTypeEnum`

```
interface InputProps = {
  label: string;
  type: InputTypeEnum;
  variant: INPUT_FIELD_VARIANTS;
  required?: boolean;
  dataName: string;
  watchedInput?: string;
  minPasswordLength?: number;
  dataTestId?: string;
};
```